fdbserver/DataDistribution.actor.cpp

```diff
@@ -3106,7 +3110,29 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 			ASSERT( self->optimalTeamCount >= 0 );
 			self->zeroOptimalTeams.set(self->optimalTeamCount == 0);
 		}
-		throw;
+		state bool sendShardsPath = false;
+		try {
+			// If server is marked as failed, a lost race condition could result in RelocateShards not being sent
+			// This is implemented as a safety net so we do not have orphaned shards sitting around
+			if (e.code() == error_code_actor_cancelled && teamContainsFailedServer(self, team)) {
+				TraceEvent("TeamTrackerStoppedSendRelocateToDDQ", self->distributorId).detail("Team", team->getDesc());
+				sendShardsPath = true;
+				teamTrackerSendRelocateShards(self, team, redundantTeam, teamContainsFailedServer(self, team), serversLeft);
+			}
+			// Trace the type of error that stopped the tracker, even if it is actor_cancelled
+			if (logTeamEvents) {
```

Comment:
This is more like a debug message than a trace event used in prod. monitoring.
The events `TeamTrackerStopping` above in the catch block already trace the team info.